jwk/handler.go

```
@@ -91,21 +92,37 @@ func (h *Handler) WellKnown(w http.ResponseWriter, r *http.Request) {
 	var jwks jose.JSONWebKeySet
 
 	for _, set := range stringslice.Unique(h.c.WellKnownKeys()) {
-		keys, err := h.r.KeyManager().GetKeySet(r.Context(), set)
-		if err != nil {
-			h.r.Writer().WriteError(w, r, err)
-			return
+		if h.c.HsmEnabled() {
+			keyPair, err := h.r.HardwareSecurityModule().FindKeyPair(nil, []byte(set))
+			if err != nil {
+				h.r.Writer().WriteError(w, r, err)
+				return
+			}
+			keyId, _ := h.r.HardwareSecurityModule().GetAttribute(keyPair, crypto11.CkaId)
```

Can we really ignore the error here?